runner/utils/driver/istio/istio_helpers.go

```diff
@@ -20,44 +20,94 @@ import (
 	"github.com/spaceuptech/space-cloud/runner/model"
 )
 
-func (i *Istio) prepareContainers(service *model.Service) []v1.Container {
+func (i *Istio) prepareContainers(service *model.Service) ([]v1.Container, []v1.Volume) {
 	// There will be n + 1 containers in the pod. Each task will have it's own container. Along with that,
 	// there will be a metric collection container as well which pushes metric data to the autoscaler.
 	// TODO: Add support for private repos
 	tasks := service.Tasks
 	containers := make([]v1.Container, len(tasks))
-	for i, task := range tasks {
-		// Prepare env variables
-		var envVars []v1.EnvVar
-		for k, v := range task.Env {
-			envVars = append(envVars, v1.EnvVar{Name: k, Value: v})
-		}
-
-		// Prepare ports to be exposed
-		ports := prepareContainerPorts(task.Ports)
-
-		// Prepare command and args
-		var cmd, args []string
-		if task.Docker.Cmd != nil {
-			cmd = task.Docker.Cmd[0:1]
-			if len(task.Docker.Cmd) > 1 {
-				args = task.Docker.Cmd[1:]
+	// Prepare Volume
+	volume := make([]v1.Volume, len(service.Secrets))
+	for j, task := range tasks {
+		for _, secretName := range service.Secrets {
```

Comment:
Wait why is everything inside this loop? That's completely wrong. You'll be adding the same container multiple times cause of this